garak/_config.py

```diff
@@ -141,6 +151,11 @@ def _load_yaml_config(settings_filenames) -> dict:
         with open(settings_filename, encoding="utf-8") as settings_file:
             settings = yaml.safe_load(settings_file)
             if settings is not None:
+                if _key_exists(settings, "api_key"):
+                    logging.info(f"API key found in {settings_filename}. Checking readability...")
+                    res = os.stat(settings_filename)
+                    if res.st_mode & stat.S_IROTH or res.st_mode & stat.S_IRGRP:
```

Comment:
One minor thought, should skip the warning if the `group` is the user?

Consider:
```
-rw-r-----  myuser domain-users
```
vs
```
-rw-r-----  myuser myuser
```

Reply:
Is there a guaranteed way of doing this? Or would the check be to skip iff groupname==username and uid==gid? Like

```
import grp
import pwd
```

then

```suggestion
                    if (res.st_mode & stat.S_IROTH) or res.st_mode & stat.S_IRGRP and not ( 
                        res.st_uid == res.st_gid and
                        grp.getgrgid(res.st_gid).gr_name == pwd.getpwuid(res.st_uid).pw_name 
                    ) :
```

Trying to think about how to engineer this even more but coming up blank -- for now. Maybe after christmas

Reply:
I'm not sure there's a way for us to do this, especially since there's no way to guarantee other users aren't in the group without enumerating those?

Reply:
As long as we are simply warning and not failing due to this check I think we can punt on further enumeration. Looking into `unix` required standards or lack of restriction this is clearly more complex than a simple equality check and `Windows` is likely worse to validate.